src/main/java/org/jenkinsci/plugins/mesos/MesosCloud.java

```diff
@@ -167,12 +172,36 @@ public String getDisplayName() {
       return "Mesos Cloud";
     }
 
-    // TODO: validate URLs
+    /** Test connection from Mesos Cloud configuration page. */
+    public FormValidation doTestConnection(@QueryParameter("mesosMasterUrl") String mesosMasterUrl) {
+      mesosMasterUrl = mesosMasterUrl.trim();
+
+      if (mesosMasterUrl.startsWith("zk://")) {
+        return FormValidation.warning("Zookeeper paths can be used, but the connection cannot be " +
+                "tested prior to saving this page.");
+      }
 
-    /** Test connection from configuration page. */
-    public FormValidation doTestConnection(
-        @QueryParameter("mesosMasterUrl") String mesosMasterUrl) {
-      throw new NotImplementedException("Connection testing is not supported yet.");
+      if (mesosMasterUrl.startsWith("http://") || mesosMasterUrl.startsWith("https://") ) {
+        return FormValidation.error("Please omit 'http(s)://'.");
```

Comment:
See above. Also, the Mesos master URL should be a valid `java.net.URL` or `java.net.URI`.

Reply:
Please checkout https://github.com/mesosphere/mesos-plugin/pull/38. `isValidUrl` should suffice.